modules/node-role-management.adoc

```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// * machine_management/managing-infrastructure-machinesets.adoc
+
+[id="node-role-management_{context}"]
+= Node role management
+
+By default, all nodes, excluding those on the control plane, are labeled with the `node-role.kubernetes.io/worker: ""` role. Nodes with just the `worker` role are called worker, or compute nodes. To create a specialized non-compute node that is not on the control plane, like an infrastructure or app node, it is recommended to keep the `worker` role and define the additional role too. For example, for an infra node, it should be labeled with the `node-role.kubernetes.io/worker: ""` and `node-role.kubernetes.io/infra: ""` roles. You can remove the `worker` role and only apply the specialized node role, but that requires creating a custom pool to allow for node upgrades due to constraints with the Machine Config Operator (MCO).
```

Comment:
I think this is the point under contention: "it is recommended to keep the `worker` role".  As @kikisdeliveryservice has pointed out, the MCO does support the ability to remove the worker role now, whereas a while back that was not the case.  
Also, just adding the label "infra" doesn't really make it a specialized non-compute node.  You've still got to either apply taints and tolerations or change your default node selector in your scheduler. I believe you already stated that above.  I might go so far as to say this whole subsection "node role management" could be removed.

Reply:
Just to reiterate @briantward point about OCP doesn't really have a concept of infra nodes - this type of documentation is more a customer courtesy - an infra pool is just another custom pool from the perspective of the MCO that must be further configured as necessary to reach whatever specific goals you have for it.